### PR TITLE
store: Implement removeAccount

### PR DESCRIPTION
**Pull request by chrisbobbe**
This PR was split out of #948, following https://github.com/zulip/zulip-flutter/pull/948#pullrequestreview-2376668035 , toward getting things merged more efficiently.

Related: #948

### CODE REVIEW

File: lib/model/store.dart [hunk 1]
```diff
@@ -850,8 +916,11 @@ class UpdateMachine {
     BackoffMachine? backoffMachine;
 
     while (true) {
+      if (_disposed) return;
+
       if (_debugLoopSignal != null) {
```
Comment (gnprice) [inline lib/model/store.dart:921]: (followup on https://github.com/zulip/zulip-flutter/pull/948#discussion_r1805674256 )

```suggestion
    assert(!_disposed);

    BackoffMachine? backoffMachine;
    while (true) {
      if (_debugLoopSignal != null) {
```

I think a check at the top of the loop shouldn't be needed — it's not in itself right after an await, and we already have checks after each await.

Conversely an assert at the top of the whole method is good for the same reason as on other methods: it helps us catch if we're attempting to call the method when the object is already disposed, which would be a bug.

**Review by gnprice — COMMENTED** [review]
Thanks! One comment below; otherwise all looks good.

File: lib/model/store.dart (hunk 1 above)
Comment (chrisbobbe) [reply lib/model/store.dart:921]: Oh indeed! Reading more closely, I see you did suggest putting it at the top of the method, not the top of the loop.

**Comment by chrisbobbe** [comment]
Thanks for the review! Revision pushed.

**Comment by gnprice** [comment]
Thanks! Looks good; merging.

**Comment by chrisbobbe** [comment]
Great! Just sent #1010 for the rest of #948.

**Pull request merged**